friction.py

```diff
@@ -127,9 +127,81 @@ def draw(self, context):
+    def check(self, context):
+        return self.basis != self.plane_hinge
     def create_entity(self):
         return Modlugre(self.name)
 
 klasses[ModlugreOperator.bl_label] = ModlugreOperator
 
+class DiscreteCoulomb(Entity):
+    def string(self):
+        string = ("discrete coulomb , " + "\n" + "\t\t\t\"" + self.links[0].name + "\", ")
+        if self.usesigma2:
+            string += " ,\n\t\t\tsigma2, " + FORMAT(self.sigma2)
+        if self.use_velocity_ratio:
+            string += " ,\n\t\t\tvelocity ratio, " + FORMAT(self.velocity_ratio)
+        if self.plane_hinge:
+            string += "simple plane hinge, " + FORMAT(self.radius)
+        else:
+            string += 'simple'
+
+        return string
+
+class DiscreteCoulombOperator(Base):
+    bl_label = "Discrete Coulomb"
+    usesigma2 = bpy.props.BoolProperty(name="Use Sigma 2", description="", default=False)
+    sigma2 = bpy.props.FloatProperty(name="Sigma 2", description="", min=0.0, max=9.9e10 , precision=6 , default=0.0)
+    use_velocity_ratio = bpy.props.BoolProperty(name="Use Velcoity Ratio", description="", default=False)
+    velocity_ratio = bpy.props.FloatProperty(name="Velocity Ratio", description="", min=0.0, max=1.0 , precision=6 , default=0.0)
+    plane_hinge = bpy.props.BoolProperty(name="Plane hinge", description="Simple plane hinge (else just simple shape function)", default=False)
+    radius = bpy.props.FloatProperty(name="Radius", description="", min=0.0, max=9.9e10, precision=6, default=0.0)
+    function_name = bpy.props.EnumProperty(items=enum_function, name="Shape Function")
+    function_edit = bpy.props.BoolProperty(name="")
+    
+    @classmethod
+    def poll(cls, context):
+        return True
+    def prereqs(self, context):
+        self.function_exists(context)
+    def assign(self, context):
+        self.entity = database.friction[context.scene.friction_index]
+        self.usesigma2 = self.entity.usesigma2
+        self.sigma2 = self.entity.sigma2
+        self.use_velocity_ratio = self.entity.use_velocity_ratio
+        self.velocity_ratio = self.entity.velocity_ratio
+        self.plane_hinge = self.entity.plane_hinge
+        self.radius = self.entity.radius
+        self.function_name = self.entity.links[0].name
+    def store(self, context):
+        self.entity = database.friction[self.index]
+        self.entity.usesigma2 = self.usesigma2
+        self.entity.sigma2 = self.sigma2
+        self.entity.use_velocity_ratio = self.use_velocity_ratio
+        self.entity.velocity_ratio = self.velocity_ratio
+        self.entity.plane_hinge = self.plane_hinge
+        self.entity.radius = self.radius
+        self.entity.unlink_all()
+        self.link_function(context, self.function_name, self.function_edit)
+        self.entity.increment_links()
+    def draw(self, context):
+        self.basis = (self.plane_hinge, self.usesigma2, self.use_velocity_ratio)
+        layout = self.layout
+        layout.prop(self, "usesigma2")
```

Comment:
Hello Doug,

So the reason I used a layout.prop instead of row.prop (as you did for modlugre) was that row.prop would append the input boxes in line with the boolean checkboxes, and if all 3 of them were checked, it would create smaller compressed boxes, where you could not see the Placeholder text ( Radius: , Velcocity Ratio : , Sigma2:  ).

That is why I opted for layout.prop, which appended them in a different line. 
Is there a problem with using layout.prop for these properties ?

Thanks for your comments. I will definitely look into them.